test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8402,30 +8408,27 @@ def cleanup_jobs_and_reservations(self, extend='forcedeletehist'):
         self.cleanup_reservations()
         return rv
 
-    def cleanup_jobs_for_perftests(self, runas=None):
+    def cleanup_large_num_jobs(self, job_ids=None, runas=None):
         """
         Helper function to cleanup jobs in performance tests that deal with
         large number of jobs. Job processes are killed manually to speed up
         cleanup in performance tests
         """
-        self.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
-        job_ids = self.select()
-
-        for jid in job_ids:
-            jobs = self.status(JOB, id=jid)
-            for job in jobs:
-                if 'session_id' in job:
-                    qstat = self.status(JOB, [ATTR_session], jid)
-                    sess_id = qstat[0][ATTR_session]
-                    self.logger.info('Killing pid [%s]' % sess_id)
-                    cmd = 'kill -9 ' + sess_id
-                    self.du.run_cmd(self.hostname, cmd, sudo=True,
-                                    runas=runas)
-            # Delete jobs from server, if any
-            try:
-                self.delete(id=jid, extend='force', wait=True, runas=runas)
-            except PbsDeleteError:
-                pass
+        status_list = self.status(JOB,
+                                  attrib=[{'job_state': 'R'}, ATTR_session])
+
+        for s in status_list:
+            if 'session_id' in s:
+                sess_id = s[ATTR_session]
+                self.logger.info('Killing pid [%s]' % sess_id)
+                cmd = 'kill -9 ' + sess_id
+                self.du.run_cmd(self.hostname, cmd, sudo=True,
+                                runas=runas)
```

Comment:
As a note, I suspect this speedup will be much slower on a multi-mom setup.  We'll only be killing the processes on MS, so the mom will still have to send signals to the other moms to kill the jobs.  I don't think we should worry about this case though.  Finding the pids of the processes on the other nodes would be really hard.